jha/src/helpers/validators.js

```diff
@@ -16,3 +21,29 @@ export const nonBCValidator = (value) => {
   return value !== 'BC'
       && value !== 'British Columbia';
 };
+
+export const dateDataRequiredValidator = (dateData) => {
+  return () => {
+    if (!dateData || !dateData.year || typeof dateData.month !== 'number' || !dateData.day) {
+      return false;
+    }
+    return true;
+  };
+};
+
+export const dateDataValidator = (dateData) => {
+  return () => {
+    if (!dateData || !dateData.year || typeof dateData.month !== 'number' || !dateData.day) {
+      return true;
+    }
+    const year = dateData.year;
+    const month = dateData.month;
+    const day = dateData.day;
+    if (!(year && typeof month === 'number' && day)
+      && (year || typeof month === 'number' || day)) {
+      return false;
```

Comment:
line 44 will never execute. If year is falsey then it has already returned.  if typeof month isn't 'number' then it has already returned, and if day is falsey then it has already returned, so in line 42/43 the if statement reads if( false && true) which translates to if(false) which will never execute.